src/gui/tray/ActivityItemContent.qml

```diff
@@ -19,19 +20,57 @@ RowLayout {
     signal dismissButtonClicked()
     signal shareButtonClicked()
 
-    spacing: 10
-
-    Image {
-        id: activityIcon
+    spacing: Style.trayHorizontalMargin
 
+    Item {
         Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
-        Layout.preferredWidth: 32
-        Layout.preferredHeight: 32
+        Layout.preferredWidth: Style.trayListItemIconSize
+        Layout.preferredHeight: Style.trayListItemIconSize
+
+        Image {
+            id: thumbnailImage
+            width: model.thumbnail.isMimeTypeIcon ? parent.width * 0.85 : parent.width * 0.8
+            height: model.thumbnail.isMimeTypeIcon ? parent.height * 0.85 : parent.height * 0.8
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.left: parent.left
+            cache: true
+            source: if(model.thumbnail !== undefined) model.thumbnail.source
```

Comment:
I still get the warnings
`
2022-03-09 12:49:43:798 [ warning default qrc:/qml/src/gui/tray/ActivityItemContent.qml:37 ]:   qrc:/qml/src/gui/tray/ActivityItemContent.qml:37:13: Unable to assign [undefined] to QUrl
2022-03-09 12:49:43:798 [ warning default qrc:/qml/src/gui/tray/ActivityItemContent.qml:33 ]:   qrc:/qml/src/gui/tray/ActivityItemContent.qml:33: TypeError: Value is undefined and could not be converted to an object
2022-03-09 12:49:43:798 [ warning default qrc:/qml/src/gui/tray/ActivityItemContent.qml:32 ]:   qrc:/qml/src/gui/tray/ActivityItemContent.qml:32: TypeError: Value is undefined and could not be converted to an object
 `
this is partially due to me having my dev nextcloud certificate being rejected by Image qml network code
still I think if we are going to access properties inside `model.thumbnail` maybe you could always create them when you return `thumbnail` role in the model